test/browser/index.js

```diff
@@ -231,91 +238,155 @@ describe(`Rendering`, () => {
         })
     })
 
-    it(`should prevent parent from DOM.selecting() inside the isolation`,
+    it(`should add a className to a vtree sink that had none`, done => {
+      function app(sources) {
+        let vtree$ = most.just(
+            h3('Hello')
+          )
+        return {
+          DOM: sources.DOM.isolateSink(vtree$, `foo`),
+        }
+      }
+      let {sources} = run(app, {
+        DOM: makeDOMDriver(createRenderTarget()),
+      })
+      // Make assertions
+      sources.DOM.select(`:root`).observable
+        .observe(root => {
+          assert.notStrictEqual(root, null)
+          assert.notStrictEqual(typeof root, `undefined`)
+          assert.strictEqual(root.tagName, `H3`)
+          assert.strictEqual(root.className, `cycle-scope-foo`)
+          done()
+        })
+    })
+
+    it(`should prevent parent from DOM.selecting inside the isolation`, done => {
+      function app(sources) {
+        return {
+          DOM: most.just(
+            h3('.top-most', [
+              sources.DOM.isolateSink(most.just(
+                div('.foo', [
+                  h4('.bar', 'Wrong')
+                ])
+              ), 'ISOLATION'),
+              h2('.bar', 'Correct')
+            ])
+          )
+        }
+      }
+      const {sources} = run(app, {
+        DOM: makeDOMDriver(createRenderTarget())
+      })
+      sources.DOM.select('.bar').observable.take(1).observe(elements => {
+        assert.strictEqual(Array.isArray(elements), true)
+        assert.strictEqual(elements.length, 1)
+        const correctElement = elements[0]
+        assert.notStrictEqual(correctElement, null)
+        assert.notStrictEqual(typeof correctElement, 'undefined')
+        assert.strictEqual(correctElement.tagName, 'H2')
+        assert.strictEqual(correctElement.textContent, 'Correct')
+        done()
+      })
+    })
+
+    it(`should prevent parent from DOM.selecting() in it's own isolation island`,
```

Comment:
This should be `should allow parent to DOM.select() in its own isolation island`.